app/src/XmlParser.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class XmlParser
+{
+  private $xml;
+
+  public function __construct($xmlEndpoint)
+  {
+    $this->xml = simplexml_load_file($xmlEndpoint);
+  }
+
+  public function getEntityDescriptors(): SimpleXMLElement
+  {
+    return $this->xml->children('md', true);
+  }
+  public function getX509Cert(SimpleXMLElement $entityDescriptor): string
+  {
+    return trim((string)$entityDescriptor
+      ->IDPSSODescriptor
+      ->KeyDescriptor
+      ->children('ds', true)
+        ->KeyInfo
+        ->X509Data
+        ->X509Certificate);
+  }
+
+  public function getSingleSignOnService(SimpleXMLElement $entityDescriptor): string
+  {
+
+    $ssos = $entityDescriptor
+    ->IDPSSODescriptor
+    ->xpath('md:SingleSignOnService');
```

Comment:
```suggestion
      ->IDPSSODescriptor
      ->xpath('md:SingleSignOnService');
```

Reply:
This indentation is intended to this the separation with `xpath`